src/test/lrsql/lrs_test.clj

```diff
@@ -211,231 +212,257 @@
-             (lrsp/-get-activity lrs auth-ident {:activityId act-1}))))
-    
-    (testing "Extremely long IRIs"
-      (is (= {:statement-ids [id-7]}
-             (lrsp/-store-statements lrs auth-ident [stmt-7] []))))
+        (testing "(single)"
+          (is (= {:statement    stmt-4
+                  :attachments  [(update-attachment-content stmt-4-attach)]}
+                 (-> (get-ss lrs
+                             auth-ident
+                             {:statementId (get stmt-4 "id") :attachments true}
+                             #{})
+                     string-result-attachment-content)))))
+
+      (testing "agent query"
+        (is (= {:person
+                {"objectType" "Person"
+                 "name" ["Sample Agent 1"]
+                 "mbox" ["mailto:sample.agent@example.com"]}}
+               (lrsp/-get-person lrs auth-ident {:agent agt-1}))))
+
+      (testing "activity query"
+        ;; Activity was updated between stmt-0 and stmt-1
+        ;; Result should contain full definition.
+        (is (= {:activity
+                {"id"         "http://www.example.com/tincan/activities/multipart"
+                 "objectType" "Activity"
+                 "definition" {"type"        "http://www.example.com/activity-types/test"
+                               "name"        {"en-US" "Multi Part Activity"
+                                              "zh-CN" "多元部分Activity"}
+                               "description" {"en-US" "Multi Part Activity Description"
+                                              "zh-CN" "多元部分Activity的简述"}}}}
+               (lrsp/-get-activity lrs auth-ident {:activityId act-1}))))
+
+      (testing "Extremely long IRIs"
+        (is (= {:statement-ids [id-7]}
+               (lrsp/-store-statements lrs auth-ident [stmt-7] []))))
+      (finally
+        (component/stop sys')
+        (support/unstrument-lrsql)))))
 
-    (component/stop sys')
-    (support/unstrument-lrsql)))
+(deftest reverse-activity-query-test
+  (let [sys   (support/test-system)
+        sys'  (component/start sys)
+        lrs   (-> sys' :lrs)
+        act-1 (get-in stmt-1 ["object" "id"])]
+    (try
+      (testing "activity query (reverse)"
+        (lrsp/-store-statements lrs auth-ident [stmt-1] [])
+        (lrsp/-store-statements lrs auth-ident [stmt-0] [])
+        ;; Activity was updated between stmt-1 and stmt-0
+        (is (= {:activity
+                {"id"         "http://www.example.com/tincan/activities/multipart"
+                 "objectType" "Activity"
+                 "definition" {"type"        "http://www.example.com/activity-types/test"
+                               "name"        {"en-US" "Multi Part Activity"
+                                              "zh-CN" "多元部分Activity"}
+                               "description" {"en-US" "Multi Part Activity Description"
+                                              "zh-CN" "多元部分Activity的简述"}}}}
```

Comment:
You should extract the activity here and in the other test case into its own def. Not only does it make the test case more concise, it also emphasizes that the two statement store orders are commutative.

Reply:
The existing activity query test (and a BUNCH of other tests) just write them out. I personally find it more readable than having to constantly check the top of the NS when working on it. That said, I can just reach into the statement for some of them so I'll do that on the code I touched, but everything else is out of scope.

Reply:
https://github.com/yetanalytics/lrsql/pull/438/commits/baee73d17490fd98a683f0126ceb7f0a44d2a343